src/main/scala/com/databricks/labs/automl/sanitize/FeatureCorrelationDetection.scala

```diff
@@ -54,30 +54,34 @@ class FeatureCorrelationDetection(data: DataFrame, fieldListing: Array[String])
   private def computeFeatureCorrelation(): Array[FeatureCorrelationStats] = {
 
     val correlationInteractions = new ArrayBuffer[FeatureCorrelationStats]
-    val redundantRecursionEliminator = new ArrayBuffer[String]
-
+   
     val taskSupport = new ForkJoinTaskSupport(new ForkJoinPool(_parallelism))
     val fieldListingPar = fieldListing.par
     fieldListingPar.tasksupport = taskSupport
 
-
-    fieldListingPar.foreach{ x =>
-      val leftFields = fieldListing.filterNot(_.contains(x)).filterNot(f => redundantRecursionEliminator.contains(f))
-      leftFields.foreach{y =>
+   //In the previous code, both of the features having higher negative or positive correlation were getting removed
+   //because for every pair of feature a and b, in correlationInteraction (a,b) as well as (b,a) was getting added
+    fieldListingPar.foreach(cfeature =>
+    {
+      var currIndex = fieldListingPar.indexOf(cfeature)
+      while (currIndex +1 < fieldListingPar.length)
+      {
+        val nextIndex = currIndex + 1
+        val nfeature = fieldsToCheck(nextIndex)
```

Comment:
Is this method defined elsewhere?  fieldsToCheck() is not present in this commit.

Reply:
In the third commit, I modified fieldsToCheck to fieldListing. By mistake, I had renamed it to fieldsToCheck.
fieldListing is Array[String] and thus fieldListing(nextIndex) will give us string at index=nextIndex